capsules/src/hmac.rs

```diff
@@ -72,55 +76,44 @@ where
         }
     }
 
-    fn run(&self) -> ReturnCode {
-        self.appid.map_or(ReturnCode::ERESERVE, move |appid| {
+    fn run(&self) -> Result<(), ErrorCode> {
+        self.appid.map_or(Err(ErrorCode::RESERVE), |appid| {
             self.apps
                 .enter(*appid, |app, _| {
-                    match app.key.as_ref() {
-                        Some(k) => {
-                            self.hmac
-                                .set_mode_hmacsha256(k.as_ref().try_into().unwrap())
-                                .unwrap();
-                        }
-                        None => {
-                            return ReturnCode::ERESERVE;
-                        }
-                    };
+                    app.key.map_or((), |k| {
```

Comment:
This looks like a possibly undesired change in functionality -- previously, if `app.key` was missing this would early-exit here with `ERESERVE`.

Reply:
@alistair23 ?